src/components/Basic/Slider/Slider.tsx

```diff
@@ -0,0 +1,176 @@
+import { ComponentProps } from "@components/types/ComponentProps";
+import { Accessor, Setter, createSignal, onMount, ParentComponent, Show, createContext, createMemo, createEffect, For } from "solid-js";
+import styles from './Slider.module.css';
+import useBaseComponent from "@components/BaseComponent/BaseComponent";
+import { clamp } from "@components/utils/clamp";
+import { Grid, SliderGrid } from "./SliderGrid";
+import { Fill, SliderFill } from "./SliderFill";
+import { Handle, SliderHandle } from "./SliderHandle";
+import { SliderThumb, Thumb } from "./SliderThumb";
+import { SliderTrack, Track } from "./SliderTrack";
+import { useToken } from "@components/utils/tokenComponents";
+
+export interface SliderRef {
+    value: Accessor<number>,
+    element: HTMLDivElement,
+    changeValue: (newValue: number) => void 
+}
+
+interface SliderProps extends ComponentProps {
+    value: number,
+    min: number,
+    max: number,
+    step: number,
+    orientation?: 'horizontal' | 'vertical',
+    onChange?: (value: number) => void;
+}
+
+interface SliderContext {
+    value: Accessor<number>,
+    percent: () => number;
+    isVertical: boolean;
+}
+
+export const SliderContext = createContext<SliderContext>();
+
+const Slider: ParentComponent<SliderProps> = (props) => {
+    const [value, setValue] = createSignal(clamp(props.value, props.min, props.max));
+    const [sliding, setSliding] = createSignal(false);
```

Comment:
I think you can omit using signal here and simply use `let` variable because the signal is not used anywhere in the jsx.

```suggestion
    let sliding = false;
```

Reply:
Resolved